alpakaConfig.h

```diff
@@ -3,58 +3,65 @@
 
 #include <alpaka/alpaka.hpp>
 
+namespace alpaka_common {
+  using Dim = alpaka::dim::DimInt<1u>;
+  using Idx = uint32_t;
+  using Extent = uint32_t;
+  using DevHost = alpaka::dev::DevCpu;
+  using PltfHost = alpaka::pltf::Pltf<DevHost>;
+  using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>;
+  using Vec = alpaka::vec::Vec<Dim, Idx>;
+}  // namespace alpaka_common
+
 #ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
-#define ALPAKA_ARCHITECTURE alpaka_cuda_async
+#define ALPAKA_ACC_GPU_CUDA_BACKEND
 namespace alpaka_cuda_async {
-  using Dim = alpaka::dim::DimInt<1u>;
-  using Idx = uint64_t;
-  using Extent = uint64_t;
+  using namespace alpaka_common;
   using Acc = alpaka::acc::AccGpuCudaRt<Dim, Extent>;
-  using DevHost = alpaka::dev::DevCpu;
   using DevAcc = alpaka::dev::Dev<Acc>;
-  using PltfHost = alpaka::pltf::Pltf<DevHost>;
   using PltfAcc = alpaka::pltf::Pltf<DevAcc>;
   using Queue = alpaka::queue::QueueCudaRtNonBlocking;
-  using WorkDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>;
-  using Vec = alpaka::vec::Vec<Dim, Idx>;
 }  // namespace alpaka_cuda_async
 
 #endif  // ALPAKA_ACC_GPU_CUDA_ENABLED
 
+#ifdef ALPAKA_ACC_GPU_CUDA_BACKEND
+#define ALPAKA_ARCHITECTURE_NAMESPACE alpaka_cuda
+#define ALPAKA_ACCELERATOR_NAMESPACE alpaka_cuda_async
```

Comment:
How would you feel about calling the namespace `ALPAKA_BACKEND_NAMESPACE`? I'm just thinking that `ACCELERATOR` sounds a bit confusing for CPU architecture. On the other hand Alpaka calls all backends as "accelerators", so maybe `ACCELERATOR` is fine.

Reply:
I picked `ALPAKA_ACCELERATOR_NAMESPACE` to mirror Cupla that uses `CUPLA_ACCELERATOR_NAMESPACE`, but I don't really mind the name.

Reply:
Would it be useful to use a consistent naming? (obviously can be changed later with minor cost, so not a big issue)

Reply:
Let's stay with the `ACCELERATOR` for now.